test/units/modules/packaging/utils.py

```diff
@@ -1,27 +1,36 @@
+import contextlib
 import json
 
 from ansible.compat.tests.mock import patch
 from ansible.module_utils import basic
 from ansible.module_utils.six.moves import xmlrpc_client
 from ansible.module_utils._text import to_bytes
 
+import pytest
+
 
 def get_method_name(request_body):
     return xmlrpc_client.loads(request_body)[1]
 
 
-def mock_request(responses, module_name):
-    def transport_request(host, handler, request_body, verbose=0):
-        """Fake request"""
-        method_name = get_method_name(request_body)
-        excepted_name, response = responses.pop(0)
-        if method_name == excepted_name:
-            if isinstance(response, Exception):
-                raise response
+@pytest.fixture
+def mock_request(mocker, responses, module_name):
+    @contextlib.contextmanager
+    def _mock_request():
+        def transport_request(host, handler, request_body, verbose=0):
+            """Fake request"""
+            method_name = get_method_name(request_body)
+            excepted_name, response = responses.pop(0)
+            if method_name == excepted_name:
+                if isinstance(response, Exception):
+                    raise response
+                else:
+                    return response
             else:
-                return response
-        else:
-            raise Exception('Expected call: %r, called with: %r' % (excepted_name, method_name))
+                raise Exception('Expected call: %r, called with: %r' % (excepted_name, method_name))
+
+        target = '{0}.xmlrpc_client.Transport.request'.format(module_name)
+        with mocker.mock_module.patch(target, side_effect=transport_request) as e:
+            yield e
 
-    target = '{0}.xmlrpc_client.Transport.request'.format(module_name)
-    return patch(target, side_effect=transport_request)
+    return _mock_request
```

Comment:
You can place fixtures into a conftest.py file and then they will be loaded automatically instead of needing an implicit import.

Reply:
It also looks like you aren't using mock_request as a fixture (telling pytest to use it by passing it to the test functions as a parameter) but just as a regular function.  Would it be better one way or the other?

Reply:
I had some difficulties because test parameters and fixture parameters where defined in the same structure. I added one commit: now `mock_request` fixture fetches required data from test parameters (`TEST_CASES`).